recipes/ixwebsocket/all/conanfile.py

```diff
@@ -47,26 +64,42 @@ def _can_use_openssl(self):
         # Older versions doesn't support OpenSSL on Mac, but those are unlikely to be built now.
         return True
 
-    def configure(self):
-        if self.settings.compiler.cppstd:
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
             # After version 11.0.8, IXWebSocket is fully compatible with C++ 11. https://github.com/machinezone/IXWebSocket/commit/ee5a2eb46ee0e109415dc02b0db85a9c76256090
             tools.check_min_cppstd(self, 14 if tools.Version(self.version) < "11.0.8" else 11)
         if self.options.tls == "applessl" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("Can only use Apple SSL on Apple.")
-        elif not self._can_use_openssl() and self.options.tls == "openssl":
+        elif not self._can_use_openssl and self.options.tls == "openssl":
             raise ConanInvalidConfiguration("This version doesn't support OpenSSL with Windows; use v7.9.3 or newer for this to be valid")
 
-    def requirements(self):
-        if self.options.get_safe("with_zlib", True):
-            self.requires("zlib/1.2.11")
-        if self.options.tls == "openssl":
-            self.requires("openssl/1.1.1j")
-        elif self.options.tls == "mbedtls":
-            self.requires("mbedtls/2.16.3-apache")
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("IXWebSocket-" + self.version, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _patch_sources(self):
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        # Use Find modules generated by conan
+        tools.replace_in_file(cmakelists,
+                              "set(CMAKE_MODULE_PATH \"${CMAKE_CURRENT_SOURCE_DIR}/CMake;${CMAKE_MODULE_PATH}\")",
+                              "list(APPEND CMAKE_MODULE_PATH \"${CMAKE_CURRENT_SOURCE_DIR}/CMake\")")
+        # Use CMake variables from FindMbedTLS.cmake generated by conan
+        tools.replace_in_file(cmakelists, "MBEDTLS_INCLUDE_DIRS", "MbedTLS_INCLUDE_DIRS")
+        tools.replace_in_file(cmakelists, "MBEDTLS_LIBRARIES", "MbedTLS_LIBRARIES")
+        # Do not force PIC
+        if tools.Version(self.version) >= "9.5.7":
+            tools.replace_in_file(cmakelists, "set(CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
+        # Allow shared
+        if tools.Version(self.version) < "11.1.4":
+            tools.replace_in_file(cmakelists, "add_library( ixwebsocket STATIC", "add_library( ixwebsocket")
+        if tools.Version(self.version) < "9.8.5":
+            tools.replace_in_file(cmakelists,
+                                  "ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib",
+                                  "ARCHIVE DESTINATION ${CMAKE_INSTALL_PREFIX}/lib LIBRARY DESTINATION lib RUNTIME DESTINATION bin")
```

Comment:
Why doesn't this install need `BUNDLE`?

Reply:
Because libraries are not BUNDLE, and this `install()` command has no executables among its TARGETS

Reply:
Oh thanks, I am an apple noob.
I don't even know what they are (or frameworks)